### PR TITLE
gh-109286: Update macOS installer to use SQLite 3.43.1

### PR DESCRIPTION
**Pull request by ghost**
This is the pr for the macOS SQLite installer upgrade for 3.43.1, following from pr #110374 where the branch was accidentally deleted.

* Issue: gh-109286

### CODE REVIEW

File: Mac/BuildScript/build-installer.py [hunk 1]
```diff
@@ -359,10 +359,10 @@ def library_recipes():
...
           ),
           dict(
-              name="SQLite 3.42.0",
-              url="https://sqlite.org/2023/sqlite-autoconf-3420000.tar.gz",
-              checksum="0c5a92bc51cf07cae45b4a1e94653dea",
-              extra_cflags=('-Os '
+              name="SQLite 3.43.1",
+              url="https://sqlite.org/2023/sqlite-autoconf-3430100.tar.gz",
+              checksum="77e61befe9c3298da0504f87772a24b0",
+	          extra_cflags=('-Os '
```
Comment (erlend-aasland) [inline Mac/BuildScript/build-installer.py]: Again, please do not include style changes.

```suggestion
              extra_cflags=('-Os '
```

File: Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Upgraded macOS SQLite installer to 3.43.1
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst]:
```suggestion
Update macOS installer to use 3.43.1.
```

**Comment by erlend-aasland** [comment]
FYI @ned-deily: I'm merging this as soon as the CI is green.

**Comment by ned-deily** [comment]
> FYI @ned-deily: I'm merging this as soon as the CI is green.

Thanks, that's fine. I'll get around to testing it soon.

File: Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Update macOS installer to use 3.43.1.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/macOS/2023-10-04-23-38-24.gh-issue-109286.1ZLMaq.rst]:
```suggestion
Update macOS installer to use SQLite 3.43.1.
```

**Pull request merged**